integreat_chat/chatanswers/services/search.py

```diff
@@ -74,4 +78,40 @@ def search_documents(
             consistency_level="Strong",
             output_fields=(["source", "text"] if include_text else ["source"])
         )[0]
-        return self.doc_details(results, include_text)
+        sources = self.doc_details(results, include_text)
+        return self.retrieve_unique_pages(sources)
```

Comment:
We should pass the MAX_RAG_DOCUMENTS here for RAG cases and the MAX_SEARCH_DOCUMENTS when we use this function to find pages with our public search endpoint.

Reply:
`results = search.search_documents(
            question,
            limit_results=settings.RAG_MAX_DOCUMENTS,
            include_text=True
        )
`

So as we are already passing the _limit_results_ argument to search_documents() which is MAX_SEARCH_DOCUMENTS by default and MAX_RAG_DOCUMENTS for the RAG, we can just pass it forward to the function?

Reply:
Okay I think I misunderstood earlier :)

We are using the _limit_results_ argument to retrieve results from the vector db (This is currently MAX_RAG_DOCUMENTS for RAG and SEARCH_MAX_DOCUMENTS for the search endpoint)

 Do you suggest introducing a new argument to filter out the unique results, or maybe re-using one of the above arguments

Reply:
Yes, my basic assumption is that we use `search_documents()` for the search endpoint AND for the document retrieval in the answer extraction (RAG) part. Both should work. For RAG, we only want 3 documents. For the search we want 10 documents (or at least another number than 3). We probably also don't need to fetch the full page content for the search. N unique links to pages should be sufficient.

Reply:
Okay makes sense :)

However, we would probably need to change the number of documents retrieved from the vector database right?

Currently, we retrieve `limit_results`(3 or 10) number of chunks from Milvus. The number of unique sources for either RAG or the search endpoint would be <= this number.

E.g - For RAG, we are currently retrieving 3 chunks from Milvus. If we want 3 unique sources from this, we need to increase the number of chunks retrieved. Or do we want to retrieve 3 chunks, and if they all point to the same page, we say that we only have 1 unique source?

Reply:
Yes, we should probably retrieve also 10 Milvus results for RAG and then reduce the amount by deduplicating results and taking only the first 3 unique pages.